native/cocos/bindings/jswrapper/v8/Utils.cpp

```diff
@@ -152,8 +154,20 @@ void setReturnValueTemplate(const Value &data, const T &argv) {
     } else if (data.getType() == Value::Type::Number) {
         argv.GetReturnValue().Set(v8::Number::New(argv.GetIsolate(), data.toDouble()));
     } else if (data.getType() == Value::Type::BigInt) {
-        // Notice: Most return value of type `size_t` should be treated as Number.
-        // argv.GetReturnValue().Set(v8::BigInt::New(argv.GetIsolate(), data.toInt64()));
+        constexpr int64_t maxSafeInt = 9007199254740991LL;  // value refer to JS Number.MAX_SAFE_INTEGER
+        constexpr int64_t minSafeInt = -9007199254740991LL; // value refer to JS Number.MIN_SAFE_INTEGER
+        if (data.toInt64() <= maxSafeInt && data.toInt64() >= minSafeInt) {
+            argv.GetReturnValue().Set(v8::Number::New(argv.GetIsolate(), static_cast<double>(data.toInt64())));
+        } else {
+            // NOTICE: We will try to convert it to BigInt, when precision loss may happend.
+            // But this will lead JS runtime exceptions such as
+            //      "Uncaught TypeError: Cannot mix BigInt and other types, use explicit conversions"
+            CC_LOG_WARNING("int64 value is out of range for double");
```

Comment:
It's better to use CC_LOG_DEBUG, otherwise game may output a lots of log in release mode which may cause performance issue.

Reply:
Using limited output in release mode